tanner/api.py

```diff
@@ -38,3 +72,48 @@ def __init__(self):
             for (i, val) in enumerate(query_res):
                 query_res[i] = json.loads(val)
         return query_res
+
+    async def return_session_info(self, sess_uuid, snare_uuid= None):
+        query_res = []
+        if snare_uuid:
+            snare_uuids = [snare_uuid]
+        else:
+            snare_uuids = await self.return_snares()
+
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            for sess in sessions:
+                if sess['sess_uuid'] == sess_uuid:
+                    return sess
+
+    async def return_sessions(self, filters, snare_uuid= None):
+        query_res = []
+        if snare_uuid:
+            snare_uuids = [snare_uuid]
+        else:
+            snare_uuids = await self.return_snares()
+
+        matching_sessions = []
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            for sess in sessions:
+                is_matching_sesssion = True
+                if 'user-agent' in filters:
+                    if filters['user-agent'] not in sess['user-agent']:
+                        is_matching_sesssion = False
+                if 'peer_ip' in filters:
+                    if filters['peer_ip'] != sess['peer_ip']:
+                        is_matching_sesssion = False
+                if 'attack_type' in filters:
+                    if filters['attack_type'] not in sess['attack_types']:
+                        is_matching_sesssion = False
+                if 'time_interval' in filters:
+                    if filters['time_interval']['end_time'] < sess['start_time'] or filters['time_interval']['start_time'] > sess['end_time']:
+                        is_matching_sesssion = False
+                if 'owner_type' in filters:
+                    if filters['owner_type'] not in sess['owner_types']:
```

Comment:
Server error when try to get owners by type. We don't have `owner_types` key in the session info
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/aiohttp/web_server.py", line 62, in handle_request
    resp = yield from self._handler(request)
  File "/usr/local/lib/python3.5/dist-packages/aiohttp/web.py", line 270, in _handle
    resp = yield from handler(request)
  File "/usr/local/lib/python3.5/dist-packages/Tanner-0.3.0-py3.5.egg/tanner/api.py", line 50, in handle_sessions
    result = await self.return_sessions(applied_filters)
  File "/usr/local/lib/python3.5/dist-packages/Tanner-0.3.0-py3.5.egg/tanner/api.py", line 146, in return_sessions
    if filters['owner_type'] not in sess['owner_types']:
KeyError: 'owner_types'
```

{'sensor_uuid': 'c6c8a20f-cc3c-490f-a0c1-4919b5e4b746', 'accepted_paths': 1, 'cookies': {'sess_uuid': None}, 'approx_time_between_requests': 0.0, 'start_time': 1489822602.9002578, **'possible_owners': ['user', 'attacker']**, 'requests_in_second': 2.384185791015625e-07, 'end_time': 1489822602.900258, 'peer_port': 53395, 'sess_uuid': '9aa2322035144fdeab7e5006e1019589', 'errors': 0, 'attack_types': [], 'peer_ip': '192.168.56.1', 'paths': [{'path': '/etc/passwd', 'timestamp': 1489822602.9002342, 'response_status': 200}], 'hidden_links': 0, 'user_agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36'}